src/utils/extractHeight.js

```diff
@@ -6,23 +11,42 @@
  */
 export async function extractHeight(ifcModel) {
   try {
-    const ifcBuildingStoreyID = 3124254112
-    const ifcBuildingStorey = ifcModel.getAllItemsOfType(ifcBuildingStoreyID, true)
+    const elevValues = []
+
+    const storeys = await ifcModel.getAllItemsOfType(IFC_TYPE_ID_BUILDING_STOREY, true)
+    const siUnits = await ifcModel.getAllItemsOfType(IFC_TYPE_ID_SI_UNIT, true)
+
+    let unitScale = 1
 
-    const printStorey = async () => {
-      const allStor = await ifcBuildingStorey
-      const elevValues = []
-      for (let i = 0; i < allStor.length; i++) {
-        elevValues[i] = allStor[i].Elevation.value
+    for (let i = 0; i < siUnits.length; i++) {
+      if (siUnits[i].UnitType.value === 'LENGTHUNIT') {
+        try {
+          const prefix = siUnits[i].Prefix.value
+          switch (prefix) {
+            case 'MILLI': unitScale = SI_PREFIX_VALUE_MILLI; break
+            case 'CENTI': unitScale = SI_PREFIX_VALUE_CENTI; break
+            default: {
+              console.warn('Unhandled length unit: ', prefix)
+              unitScale = 1
+            }
+          }
+        } catch {
+          console.warn('No length unit found')
+        }
       }
-      return elevValues
     }
-    const elevValues = []
-    for (let i = 0; i < ifcBuildingStorey.length; i++) {
-      elevValues[i] = ifcBuildingStorey[i].Elevation.value
+
+    for (let i = 0; i < storeys.length; i++) {
+      const elevation = storeys[i].Elevation.value
+      if (!isFinite(elevation)) {
+        console.warn('Found invalid elevation value: ', elevation)
+        continue
+      }
+      elevValues[i] = elevation * unitScale
     }
-    return await printStorey()
+
+    return elevValues
   } catch {
-    console.log('No Levels detected')
+    console.log('No levels found')
```

Comment:
@pablo-mayrgundter please note`console.warn` caused test failures, hence I changed to `console.log`. I believe the reason of the test failures is that the ifcmodel is often loaded multiple times, before the calculation of the extract levels take place.